src/azure-cli-core/azure/cli/core/extension/dynamic_install.py

```diff
@@ -121,7 +121,14 @@ def try_install_extension(parser, args):
 
 
 def _get_extension_use_dynamic_install_config(cli_ctx):
-    default_value = 'yes_prompt'
+    from knack.prompting import verify_is_a_tty, NoTTYException
+    try:
+        verify_is_a_tty()
```

Comment:
Not that `sys.stdin.isatty()` may not be accurate. For example, if a command in run in terminal but redirected, `sys.stdin.isatty()` returns `False`:

```
$ python -c "import sys; print(sys.stdin.isatty())"
True

$ echo "test" | python -c "import sys; print(sys.stdin.isatty())"
False
```